pkg/kube/multicluster/secretcontroller_test.go

```diff
@@ -94,8 +95,53 @@ func resetCallbackData() {
 	deleted = ""
 }
 
+func Test_KubeConfigOverride(t *testing.T) {
+	var (
+		expectedQPS   = float32(100)
+		expectedBurst = 200
+	)
+	fakeRestConfig := &rest.Config{}
+	BuildClientsFromConfig = func(kubeConfig []byte, c cluster.ID, configOverrides ...func(*rest.Config)) (kube.Client, error) {
+		for _, override := range configOverrides {
+			override(fakeRestConfig)
+		}
+		return kube.NewFakeClient(), nil
+	}
+	clientset := kube.NewFakeClient()
+	stopCh := test.NewStop(t)
+	c := NewController(clientset, secretNamespace, "", mesh.NewFixedWatcher(nil), func(cfg *rest.Config) {
+		cfg.QPS = expectedQPS
+		cfg.Burst = expectedBurst
+	})
+	clientset.RunAndWait(stopCh)
+	c.AddHandler(&handler{})
+	clientset.RunAndWait(stopCh)
+	_ = c.Run(stopCh)
+	t.Run("sync timeout", func(t *testing.T) {
+		retry.UntilOrFail(t, c.HasSynced, retry.Timeout(2*time.Second))
+	})
+	kube.WaitForCacheSync("test", stopCh, c.HasSynced)
+	secret0 := makeSecret(secretNamespace, "s0",
+		clusterCredential{"c0", []byte("kubeconfig0-0")})
```

Comment:
I guess these ops are not needed

Reply:
It will not create a cluster client if the data of secret is empty and the overrideFunc will not take effect.